src/controllers/profile.controller.js

```diff
@@ -0,0 +1,123 @@
+import { ApiError } from '../exceptions/api.error.js';
+import { userService } from '../services/user.service.js';
+import { User } from '../models/user.js';
+import bcrypt from 'bcrypt';
+import { validatePassword } from '../utils/methods.js';
+import { emailService } from '../services/emai.services.js';
+import jwt from 'jsonwebtoken';
+
+const rename = async (req, res) => {
+  const { newName, email } = req.body;
+
+  if (!newName) {
+    throw ApiError.badRequest('Bad request, need to provide new name of user');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  await User.update({ name: newName }, { where: { id: user.id } });
+
+  res.status(200).send('Done');
+};
+
+const changePassword = async (req, res) => {
+  const { email, password, newPassword, confirmationPassword } = req.body;
+  const user = await userService.getByEmail(email);
```

Comment:
There is no check to see if the user exists after fetching by email. You should add a check to ensure the user is found before proceeding with password validation.